src/Altinn.Dan.Plugin.DATASOURCENAME/Main.cs

```diff
@@ -119,9 +123,10 @@ private async Task<List<EvidenceValue>> GetEvidenceValuesDatasetName2(EvidenceHa
             FunctionContext context)
         {
             _logger = context.GetLogger(context.FunctionDefinition.Name);
-            _logger.LogInformation($"Running metadata for {Constants.EvidenceSourceMetadataFunctionName}");
+            _logger.LogInformation($"Running func metadata for {Constants.EvidenceSourceMetadataFunctionName}");
             var response = req.CreateResponse(HttpStatusCode.OK);
-            await response.WriteAsJsonAsync(_metadata.GetEvidenceCodes());
+            await response.WriteAsJsonAsync(_evidenceSourceMetadata.GetEvidenceCodes());
```

Comment:
This will fail to serialize authorization requirements, which require type information to be included. For this we need to supply a custom JSON.Net serializer:

`await response.WriteAsJsonAsync(new Metadata().GetEvidenceCodes(), new NewtonsoftJsonObjectSerializer(new JsonSerializerSettings() { TypeNameHandling = TypeNameHandling.Auto}));`

This is of course a bit too much boilerplate, so the whole EvidenceSourceResponse-class needs to be reworked in order to support .NET6, as IActionResult cannot be used. We should probably look into splitting Common into enums/models and utils, where the latter is available for both Core 3.1 and .NET6 (using IActionResult and HttpResponseData/WriteAsJsonAsync respectively). This can probably be done in a separate PR though.

Reply:
Added serializer.
I've noticed that EvidenceSourceResponse needs some work.. 
The split sounds like a good idea, would make the transition to .NET6 easier.